src/MOI_wrapper.jl

```diff
@@ -0,0 +1,225 @@
+import MathOptInterface as MOI
+using ManifoldsBase: ManifoldsBase
+using Manifolds: Manifolds
+
+include("qp_block_data.jl")
+
+const _FUNCTIONS = Union{
+    MOI.VariableIndex,MOI.ScalarAffineFunction{Float64},MOI.ScalarQuadraticFunction{Float64}
+}
+
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+function MOI.dimension(::VectorizedManifold{<:Manifolds.Sphere{N}}) where {N}
+    return N + 1
+end
+
+struct _EmptyNLPEvaluator <: MOI.AbstractNLPEvaluator end
+MOI.features_available(::_EmptyNLPEvaluator) = [:Grad, :Jac, :Hess]
+MOI.initialize(::_EmptyNLPEvaluator, ::Any) = nothing
+
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    num_variables::Int
```

Comment:
This should really be the dimension of the representation so it should be `prod(representation_size(manifold))`

Reply:
ok, before calling any function within `Manopt` one would have to reshape that back to the `representation_size` then. And it does not work with all manifolds, for example not with `FixedRankMatrices` or with the Poincare Halfplane or the Poincare ball representations of `Hyperbolic`, since those are their own structs and not vectors.

Also – the dimension of the manifold (i.e. the actual degrees of freedom) is still just `manifold_dimension`, which is usually (at least a bit sometimes even about a factor 1/2) smaller than the representation.

Reply:
We can try with `prod(representation_size())` and we can switch to something more general afterwards if we want to make it work for more manifolds. It just has to go back and forth in vectorization mode with `JuMP.vectorize` and `JuMP.reshape`

Reply:
Yes, and the vectorise should cover most for now – the others (e.g. the ones I mentioned) just probably need their own vectorize and “devectorize” methods then (maybe overloading vectorise and reshape then).
But having an INterface that works for matrices/vectors would be even the main step I think.